frontend/src/components/NewsComponent.jsx

```diff
@@ -8,9 +8,23 @@ const NewsComponent = ({
   setArticleModalData,
   uiData,
   setMetaData,
+  handleSignalUpdates,
 }) => {
-  const openModal = () => {
-    setArticleModalData(article.articleURL); // just so the modal appears on the screen
+  // enums
+  const SignalTypes = {
+    OPEN: "open",
+    LIKED: "liked",
+  };
```

Comment:
SignalTypes should be one const enum thats used in both places, maybe extract it into a seperate util file and import it in both NewsComponent and ArticleModal